### PR TITLE
DM-43942: Serialize fgcm QA plots to the Butler

### PR DESCRIPTION
**Pull request by erykoff**

### CODE REVIEW

**Comment by laurenam** [comment]
Quick initial comment:  I think you now need to update the `cookbook` to describe the plot persistence.

File: python/lsst/fgcmcal/fgcmFitCycle.py [hunk 1]
```diff
@@ -317,18 +317,21 @@ def __init__(self, *, config=None):
...
+                outputConnections = copy.copy(inputOutputConnections)
+                outputConnections.extend(multicycleOutputConnections)
+                if config.outputZeropointsBeforeFinalCycle:
+                    outputConnections.extend(optionalZpOutputConnections)
+                if config.outputStandardsBeforeFinalCycle:
+                    outputConnections.extend(optionalStarOutputConnections)
+
+                if config.doPlots:
+                    if cycle == (config.multipleCyclesFinalCycleNumber - 1) \
+                       or config.doPlotsBeforeFinalCycles:
```
[the remark this answers is not in the document]
Comment (erykoff) [reply python/lsst/fgcmcal/fgcmFitCycle.py:332]: Yes, I've added more comments.

File: python/lsst/fgcmcal/fgcmFitCycle.py [hunk 2]
```diff
@@ -196,8 +341,10 @@ def __init__(self, *, config=None):
             outputConnections = copy.copy(inputOutputConnections)
             outputConnections.extend(optionalZpOutputConnections)
             outputConnections.extend(optionalStarOutputConnections)
+            if config.doPlots:
+                outputConnections.extend(plotConnections)
             for (name, storageClass, doc) in outputConnections:
-                connectionName = f"{name}{config.multipleCyclesFinalCycleNumber}"
+                connectionName = f"fgcm_Cycle{config.multipleCyclesFinalCycleNumber}_{name}"
```
[the remark this answers is not in the document]
Comment (erykoff) [reply python/lsst/fgcmcal/fgcmFitCycle.py:355]: As we discussed on slack, I have found over time that having all the plots of a given cycle sorted together is more useful than having plots of the same type from different cycles sorted together.

File: python/lsst/fgcmcal/fgcmFitCycle.py [hunk 3]
```diff
@@ -148,95 +148,101 @@ class FgcmFitCycleConnections(pipeBase.PipelineTaskConnections,
...
+                object.__setattr__(self, connectionName, outConnection)
+                self.outputs.add(connectionName)
+        else:
+            # Single cycle run.
+            if config.cycleNumber > 0:
+                inputConnections = copy.copy(inputOutputConnections)
+            else:
+                inputConnections = []
+            outputConnections = copy.copy(inputOutputConnections)
+            if config.isFinalCycle or config.outputZeropointsBeforeFinalCycle:
```
[the remark this answers is not in the document]
Comment (erykoff) [reply python/lsst/fgcmcal/fgcmFitCycle.py:372]: I've added a comment as to why we want these.

File: python/lsst/fgcmcal/fgcmFitCycle.py [hunk 4]
```diff
@@ -232,8 +382,11 @@ def __init__(self, *, config=None):
                 self.prerequisiteInputs.add(connectionName)
 
             for (name, storageClass, doc) in outputConnections:
-                connectionName = name
-                storageName = f"{name}{config.cycleNumber}"
+                connectionName = f"fgcm{name}"
+                storageName = f"fgcm_Cycle{config.cycleNumber}_{name}"
```
[the remark this answers is not in the document]
Comment (erykoff) [reply python/lsst/fgcmcal/fgcmFitCycle.py:394]: The variables/connections here are all old-school `camelCase`.  The storage classes follow patterns from other plot types, 🤷

**Review by laurenam — APPROVED** [review]

File: doc/lsst.fgcmcal/tasks/lsst.fgcmcal.fgcmBuildFromIsolatedStars.FgcmBuildFromIsolatedStarsTask.rst [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+The first is :doc:`lsst.fgcmcal.fgcmMakeLut.FgcmMakeLutTask`, the third is :doc:`lsst.fgcmcal.fgcmFitCycle.FgcmFitCycleTask`, and the fourth is :doc:`lsst.fgcmcal.fgcmOutputProducts.FgcmOutputProductsTask`.
+
+.. _lsst.fgcmcal.fgcmBuildStars.FgcmBuildFromIsolatedStarsTask-summary:
+
+Processing summary
+==================
+
+``FgcmBuildFromIsolatedStarsTask`` runs this sequence of operations:
+
+#. Finds unique visits and collates visit metadata, including exposure time, pointing, typical psf size, background level.
```
[the remark this answers is not in the document]
Comment (erykoff) [reply doc/lsst.fgcmcal/tasks/lsst.fgcmcal.fgcmBuildFromIsolatedStars.FgcmBuildFromIsolatedStarsTask.rst:20]: A gen2ism that is now removed.

**Pull request merged**